atom/browser/net/atom_url_loader_factory.cc

```diff
@@ -216,6 +218,43 @@ void AtomURLLoaderFactory::StartLoading(
     }
   }
 
+  network::ResourceResponseHead head = ToResponseHead(dict);
+
+  // Handle redirection.
+  //
+  // Note that with NetworkService, sending the "Location" header no longer
+  // automatically redirects the request, we have explicitly create a new loader
+  // to implement redirection. This is also what Chromium does with WebRequest
+  // API in WebRequestProxyingURLLoaderFactory.
+  std::string location;
+  if (head.headers->IsRedirect(&location)) {
+    network::ResourceRequest new_request = request;
+    new_request.url = GURL(location);
+    // When the redirection comes from an intercepted scheme (which has
+    // |proxy_factory| passed), we askes the proxy factory to create a loader
+    // for new URL, otherwise we call |StartLoadingHttp|, which creates
+    // loader with default factory.
+    //
+    // Note that when handling requests for intercepted scheme, creating loader
+    // with default factory (i.e. calling StartLoadingHttp) would bypass the
+    // ProxyingURLLoaderFactory, we have to explicitly use the proxy factory to
+    // create loader so it is possible to have handlers of intercepted scheme
+    // getting called recursively, which is a behavior expected in protocol
+    // module.
+    //
+    // I'm not sure whether this is an intended behavior in Chromium.
```

Comment:
I think these are fine with our current implementation of `ProxyingURLLoaderFactory`, but when we redesign to an implementation that uses one `AtomURLLoaderFactory` per network context, we would have to close the current loader connection , so that the redirect request will create a new one and keeps the right reference count of the class.